ringdown/fit.py

```diff
@@ -762,10 +742,10 @@ def run(self, prior=False, suppress_warnings=True, store_residuals=True,
                             kws['draws'] = draws
 
                             self.run(prior=prior, suppress_warnings=suppress_warnings, store_residuals=store_residuals, min_ess=min_ess, **kws)
-
-        if store_residuals:
-            self._generate_whitened_residuals()
-    run.__doc__ = run.__doc__.format(DEF_RUN_KWS, DEF_RUN_PRIOR_KWS)
+        if not prior:
```

Comment:
can you simplify these nested if statements? `if store_residuals and not prior:`

Reply:
Done, see commit.